internal/shader/syntax_test.go

```diff
@@ -1314,6 +1314,149 @@ func Fragment(dstPos vec4, srcPos vec2, color vec4) vec4 {
 	}
 }
 
+// Issue: #2755
+func TestSyntaxOperatorShift(t *testing.T) {
+	cases := []struct {
+		stmt string
+		err  bool
+	}{
+		{stmt: "a := 1 << 2; _ = a", err: false},
```

Comment:
Does this test work?

```go
{stmt: "a := 1.0 << 2.0; _ = a", err: false},
```

Reply:
No, it doesn't. In `ResolveUntypedConstsForBinaryOp` it is resolved to `(float << float)`.
The same behavior could be observed with other binary operators, such as `^`, `&`, and  `|`.
However, op + assignment can truncate to integer, meaning `a := 1; a <<= 2.0; _ = a` is valid.

I will fix that if it is expected not to alert an error in `a := 1 << 2.0` case.

Reply:
These are expected to work without errors. Please fix them. Thanks!

https://go.dev/play/p/ZBrITUbJjlN

Reply:
The interesting thing is, a strict type checking for constants is valid for `%`, but not for `<<` or `>>`.

https://go.dev/play/p/kF_j7sWFfCU

Reply:
Today I found even `go vet` failed to treat this correctly! https://github.com/golang/go/issues/65939